src/NSubstitute.Analyzers.VisualBasic/DiagnosticAnalyzers/ArgumentMatcherCompilationAnalyzer.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using NSubstitute.Analyzers.Shared;
+using NSubstitute.Analyzers.Shared.DiagnosticAnalyzers;
+
+namespace NSubstitute.Analyzers.VisualBasic.DiagnosticAnalyzers
+{
+    internal sealed class ArgumentMatcherCompilationAnalyzer : AbstractArgumentMatcherCompilationAnalyzer<InvocationExpressionSyntax, MemberAccessExpressionSyntax, ArgumentSyntax>
+    {
+        private static ImmutableArray<ImmutableArray<int>> AllowedPaths { get; } = ImmutableArray.Create(
```

Comment:
These seem duplicated with the CSharp equivalent. Maybe move to base class and share the values?

Reply:
Unfortunately, we cant. SyntaxKind is defined separately for VisualBasic and CSharp and the values of corresponding enums don't match. For instance
C#
```c#
ArgumentList = 8636, // 0x21BC
```
VB
```vb
ArgumentList = 347, // 0x015B
```